preview/MsixCore/MsixCoreInstaller/ValidateTargetDeviceFamily.cpp

```diff
@@ -58,33 +45,50 @@ HRESULT ValidateTargetDeviceFamily::ParseTargetDeviceFamilyFromPackage()
         return HRESULT_FROM_WIN32(ERROR_INSTALL_REJECTED);
     }
 
-    ComPtr<IMsixElement> dependencyElement;
-    RETURN_IF_FAILED(dependencyEnum->GetCurrent(&dependencyElement));
-
-    Text<wchar_t> targetDeviceFamilyName;
-    RETURN_IF_FAILED(dependencyElement->GetAttributeValue(L"Name", &targetDeviceFamilyName));
-    m_targetDeviceFamilyName = targetDeviceFamilyName.Get();
-
-    Text<wchar_t> minVersion;
-    RETURN_IF_FAILED(dependencyElement->GetAttributeValue(L"MinVersion", &minVersion));
-    std::wstring manifestMinVersion = minVersion.Get();
-
-    /// Major version
-    size_t start = 0;
-    size_t end = manifestMinVersion.find_first_of(L'.');
-    m_majorVersion = manifestMinVersion.substr(start, end - start);
-
-    /// Minor version
-    manifestMinVersion.replace(start, end - start + 1, L"");
-    end = manifestMinVersion.find_first_of(L'.');
-    m_minorVersion = manifestMinVersion.substr(start, end - start);
+    while (hc)
+    {
+        ComPtr<IMsixElement> dependencyElement;
+        RETURN_IF_FAILED(dependencyEnum->GetCurrent(&dependencyElement));
+
+        Text<wchar_t> targetDeviceFamilyName;
+        RETURN_IF_FAILED(dependencyElement->GetAttributeValue(L"Name", &targetDeviceFamilyName));
+        m_targetDeviceFamilyName = targetDeviceFamilyName.Get();
+
+        Text<wchar_t> minVersion;
+        RETURN_IF_FAILED(dependencyElement->GetAttributeValue(L"MinVersion", &minVersion));
+        std::wstring manifestMinVersion = minVersion.Get();
+
+        /// Major version
+        size_t start = 0;
+        size_t end = manifestMinVersion.find_first_of(L'.');
+        m_majorVersion = manifestMinVersion.substr(start, end - start);
+
+        /// Minor version
+        manifestMinVersion.replace(start, end - start + 1, L"");
+        end = manifestMinVersion.find_first_of(L'.');
+        m_minorVersion = manifestMinVersion.substr(start, end - start);
+
+        /// Build number
+        manifestMinVersion.replace(start, end - start + 1, L"");
+        end = manifestMinVersion.find_first_of(L'.');
+        m_buildNumber = manifestMinVersion.substr(start, end - start);
+
+        /// Return if any one of the target device families are compatible with OS
+        if (IsTargetDeviceFamilyNameCompatibleWithOS() && IsManifestVersionCompatibleWithOS())
+        {
+            TraceLoggingWrite(g_MsixTraceLoggingProvider,
+                "Target device family name and manifest min version are compatible with OS",
+                TraceLoggingLevel(WINEVENT_LEVEL_INFO));
+            return S_OK;
+        }

```

Comment:
>} [](start = 8, length = 1)

I think it would be helpful to add a tracelogging to tell us what values we found in the manifest that were not compatible.

Reply:
Added logging for values in either case.